### PR TITLE
Logger added. Small sugar fixes.

### PR DESCRIPTION
**Pull request by wjdda**
1) Log4j Logger.
 ++ gamelogger/LogConfFactory. Configurating root logger (for every new
 loggers in classes).
 ++ gamelogger/LogOptions. Constants to simple configurating loggers.

 2) Other classes:
 <> printMsg -> LOG.
 <> Some small code corrections.
 ++ Added comments "QUESTION", "TEST" (For IDE marker parser).

 3) Main.
 -- printMsg().
 ++ initLogger(). Configure root logger for future children generation.
 ++ initControllers(). Logger in MouseController creating earlier than
 root logger. Moved initialization here.

### CODE REVIEW

**Comment by wjdda** [comment]
Don't forget project in Trello.

File: src/com/company/gamelogger/LogConfFactory.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+@Plugin(name = "LogConfFactory", category = ConfigurationFactory.CATEGORY)
+@Order(50)
+public class LogConfFactory extends ConfigurationFactory {
+    private static Configuration createConfiguration(final String name, ConfigurationBuilder<BuiltConfiguration> builder) {
+        builder.setConfigurationName(name);
+        builder.setStatusLevel(LEVEL);
+
+        String logFilePath = FILE_PATH + "/" + FILE_NAME;
+        boolean debugging = LEVEL.equals(Level.DEBUG) || LEVEL.equals(Level.ALL);
```
Comment (AlexanderSamoilov) [inline src/com/company/gamelogger/LogConfFactory.java]: This is more correct:
boolean debugging = LEVEL.getlevel() <= Level.DEBUG;

File: src/com/company/gamelogger/LogOptions.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    // Layout instructions:
+    //    https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html
+    //    http://logging.apache.org/log4j/2.x/manual/layouts.html
+    //
+    // WARNING. Keys: %c, %l, %M do program's speed very slow!
+
+    // Example of good format of output:
+    //    2019-04-16 22:06:54 [22] gamethread.Main > mainfunc() INFO: The game starts.
+    static final String STDOUT_LAYOUT = "%d{yyyy-MM-dd HH:mm:ss} [%-2.2tid] %-30.30c{2} > %20.20M() %5level: %msg%n%throwable";
```
Comment (AlexanderSamoilov) [inline src/com/company/gamelogger/LogOptions.java]: I propose to use just %tid. You never know what numbers can Java threads have. It is very OS-dependent.
I saw many times on Linux the output of "jmap" and it showed me bigger numbers.
The longest which I saw was always 5 digits which makes me think that the maximal thread ID is _probably_ 32767, however I am not sure.
That is why I propose to not predict it at let your OS to write the comlpete thread ID without making it cut.

File: src/com/company/gamecontent/GameMap.java [hunk 3]
```diff
@@ -374,9 +391,9 @@ public boolean occupied(Rectangle givenRect, GameObject exceptObject) {
                     Rectangle objOnBlockRect = objOnBlock.getRect();
 
                     // DEBUG
-                    //Main.printMsg("Check 1: (" + thisObjRect.x + "," + thisObjRect.y + "," + thisObjRect.width + "," + thisObjRect.height);
-                    //Main.printMsg("Check 2: (" + objOnBlockRect.getX() + "," + objOnBlockRect.getY() + "," + objOnBlockRect.getWidth() + "," + objOnBlockRect.getHeight());
-                    //Main.printMsg("Check 3: (" + objOnBlock.getAbsLoc()[0] + "," + objOnBlock.getAbsLoc()[1] + "," + objOnBlock.getAbsSize()[0] + "," + objOnBlock.getAbsSize()[1]);
+//                    LOG.debug("Check 1: (" + thisObjRect.x + "," + thisObjRect.y + "," + thisObjRect.width + "," + thisObjRect.height);
```
Comment (AlexanderSamoilov) [inline src/com/company/gamecontent/GameMap.java:394]: I see you forgot to answer my comment about it.
I asked you why did you change function calls inside these 3 commented lines. I suppose that you wrongly use the updated API from your local workspace and did not revert it before doing commit.

File: src/com/company/gamelogger/LogConfFactory.java (hunk 1 above)
Comment (wjdda) [reply src/com/company/gamelogger/LogConfFactory.java]: See new commit!

File: src/com/company/gamelogger/LogOptions.java (hunk 2 above)
Comment (wjdda) [reply src/com/company/gamelogger/LogOptions.java]: Added 10 digits to tid. If we don't do this, the layout breaks down.

File: src/com/company/gamecontent/GameMap.java (hunk 3 above)
Comment (wjdda) [reply src/com/company/gamecontent/GameMap.java:394]: Now I have done a log version 2.x import everywhere.
